src/main/groovy/com/lesfurets/jenkins/unit/BasePipelineTest.groovy

```diff
@@ -25,13 +25,30 @@ abstract class BasePipelineTest {
 
     def stringInterceptor = { m -> m.variable }
 
+    def usernamePasswordInterceptor = { m -> [m.usernameVariable, m.passwordVariable] }
+
     def withCredentialsInterceptor = { list, closure ->
-        list.forEach {
-            binding.setVariable(it, "$it")
+        list.forEach { creds ->
+            if (creds instanceof String) {
+                binding.setVariable(creds, creds)                
+            } else {
+                creds.each {
+                    def var = it.toString()
+                    binding.setVariable(var, var)
+                }
+            }
         }
+        closure.delegate = delegate
         def res = closure.call()
```

Comment:
Could you please use
```
helper.callClosure(closure)
```

Reply:
OK, I will do so. I used what was already there by the way.

Reply:
Done, now using callClosure